crates/mcp-client/src/transport/logging/file_logger.rs

```diff
@@ -0,0 +1,54 @@
+use std::fs::{File, OpenOptions};
+use std::io::Write;
+use std::path::PathBuf;
+use std::sync::Arc;
+use tokio::sync::Mutex;
+use chrono::Local;
+
+use super::LogMessage;
+
+pub struct FileLogger {
```

Comment:
nit: i wonder if it would be more consistent/easier to forward to `tracing`?  please disregard if that adds a lot of complexity, but off the top of my head it might make it easier to control both file and stderr based logging etc